modules/caddytls/automation.go

```diff
@@ -183,8 +183,14 @@ func (ap *AutomationPolicy) Provision(tlsApp *TLS) error {
 		ap.Issuer = val.(certmagic.Issuer)
 	}
 
+	keyType := ap.KeyType
+	if keyType != "" {
+		repl := caddy.NewReplacer()
+		keyType = repl.ReplaceAll(ap.KeyType, "")
+	}
+
 	keySource := certmagic.StandardKeyGenerator{
-		KeyType: supportedCertKeyTypes[ap.KeyType],
+		KeyType: supportedCertKeyTypes[keyType],
```

Comment:
Since you're setting the key type here, would be best to do the validation at this point, too (probably no need to use the Validate() method in your case)